lib/vmpooler/api/v1.rb

```diff
@@ -509,6 +514,47 @@ def need_token!
 
       JSON.pretty_generate(result)
     end
+
+    post "#{api_prefix}/vm/:hostname/snapshot/?" do
+      content_type :json
+
+      status 404
+      result = { 'ok' => false }
+
+      params[:hostname] = hostname_shorten(params[:hostname], config['domain'])
+
+      if backend.exists('vmpooler__vm__' + params[:hostname])
+        result[params[:hostname]] = {}
+
+        o = [('a'..'z'), ('0'..'9')].map(&:to_a).flatten
+        result[params[:hostname]]['snapshot'] = o[rand(25)] + (0...31).map { o[rand(o.length)] }.join
```

Comment:
Would definitely say this pair of lines is a candidate for extracting to a method with an intention-revealing name.

``` ruby
  def random_snapshot_name
    o = [('a'..'z'), ('0'..'9')].map(&:to_a).flatten
    o[rand(25)] + (0...31).map { o[rand(o.length)] }.join
  end

  # ...

    post "#{api_prefix}/vm/:hostname/snapshot/?" do
      content_type :json

      status 404
      result = { 'ok' => false }

      hostname = params[:hostname] = hostname_shorten(params[:hostname], config['domain'])

      if backend.exists("vmpooler__vm__#{hostname}")
        result[hostname] = {}
        result[hostname]['snapshot'] = random_snapshot_name
        backend.sadd('vmpooler__tasks__snapshot', hostname + ':' + result[hostname]['snapshot'])

        status 202
        result['ok'] = true
      end

      JSON.pretty_generate(result)
    end
```